### PR TITLE
add server

### PR DESCRIPTION
**Pull request by Anton-Zhytonbaiev**

### CODE REVIEW

File: src/createServer.js [hunk 1]
```diff
@@ -1,3 +1,36 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
```
Comment (OleksandrProkop) [inline src/createServer.js:2]: enough ./convertToCase

File: src/createServer.js [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+    res.setHeader('Content-Type', 'application/json');
+
+    const normalizeURL = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = normalizeURL.pathname.slice(1);
+    const targetCase = normalizeURL.searchParams.get('toCase');
+    const errors = validation(originalText, targetCase);
+
+    if (errors.errors.length) {
+      req.statusCode = 404;
+      req.statusMessage = 'Bad request';
```
Comment (OleksandrProkop) [inline src/createServer.js:16]: i think we should respond with 400 status, not request

**Review by OleksandrProkop — APPROVED** [review]
gj

File: src/validation.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    });
+  }
+
+  if (!supportedCases.includes(targetCase) && targetCase) {
+    allErrors.push({
+      message: errorMessage.notSupportedCase,
+    });
+  }
+
+  return { errors: allErrors };
```
Comment (alexandra-protyanova) [inline src/validation.js:25]: I think it is better to return array of errors

File: src/createServer.js (hunk 2 above)
Comment (alexandra-protyanova) [reply src/createServer.js:16]: agree

**Review by alexandra-protyanova — CHANGES_REQUESTED** [review]

**Pull request closed without merging**